### PR TITLE
Create Property column to list properties

### PR DESCRIPTION
**Pull request by chngchngchng**
As we are including functionality to store properties that an agent is in charge of, let's create a separate column next to the users that will display the properties stored in Cobb.

The internal implementation of the property list is the same as that of the person list, so most commands that work on the person list can be adapted to work on the property list with minimal difficulty.

### CODE REVIEW

File: src/main/java/seedu/address/model/AddressBook.java [hunk 1]
```diff
@@ -50,13 +54,22 @@ public void setPersons(List<Person> persons) {
         this.persons.setPersons(persons);
     }
 
+    /**
+     * Replaces the contents of the property list with {@code properties}.
+     * {@code persons} must not contain duplicate persons.
```
Comment (jchilling) [inline src/main/java/seedu/address/model/AddressBook.java:59]:
```suggestion
     *{@code properties} must not contain duplicate properties.
```

File: src/main/java/seedu/address/model/AddressBook.java [hunk 2]
```diff
@@ -77,6 +90,14 @@ public void addPerson(Person p) {
         persons.add(p);
     }
 
+    /**
+     * Adds a person to the address book.
+     * The person must not already exist in the address book.
```
Comment (jchilling) [inline src/main/java/seedu/address/model/AddressBook.java:95]:
```suggestion
     * Adds a property to the address book.
     * The property must not already exist in the address book.
```

File: src/main/java/seedu/address/model/AddressBook.java [hunk 3]
```diff
@@ -128,6 +150,12 @@ public ObservableList<Person> getPersonList() {
         return persons.asUnmodifiableObservableList();
     }
 
+    @Override
+    public ObservableList<Property> getPropertyList() {
+        System.out.println(properties.asUnmodifiableObservableList().size());
```
Comment (jchilling) [inline src/main/java/seedu/address/model/AddressBook.java:155]: Is this for testing? Might need to delete it later to keep the code neat.

File: src/main/java/seedu/address/ui/PropertyCard.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    private Label price;
+    @FXML
+    private Label address;
+    @FXML
+    private Label description;
+    @FXML
+    private FlowPane tags;
+
+    /**
+     * Creates a {@code PersonCode} with the given {@code Person} and index to display.
```
Comment (jchilling) [inline src/main/java/seedu/address/ui/PropertyCard.java:47]:
```suggestion
     * Creates a {@code PropertyCard} with the given {@code Property} and index to display.
```

File: src/main/java/seedu/address/ui/PropertyListPanel.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+import javafx.fxml.FXML;
+import javafx.scene.control.ListCell;
+import javafx.scene.control.ListView;
+import javafx.scene.layout.Region;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.model.property.Property;
+
+
+/**
+ * Panel containing the list of persons.
```
Comment (jchilling) [inline src/main/java/seedu/address/ui/PropertyListPanel.java:15]:
```suggestion
 * Panel containing the list of properties.
```

File: src/main/java/seedu/address/ui/PropertyListPanel.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+ */
+public class PropertyListPanel extends UiPart<Region> {
+    private static final String FXML = "PropertyListPanel.fxml";
+    private final Logger logger = LogsCenter.getLogger(PropertyListPanel.class);
+
+    @FXML
+    private ListView<Property> propertyListView;
+
+    /**
+     * Creates a {@code PersonListPanel} with the given {@code ObservableList}.
```
Comment (jchilling) [inline src/main/java/seedu/address/ui/PropertyListPanel.java:25]:
```suggestion
     * Creates a {@code PropertyListPanel} with the given {@code ObservableList}.
```

File: src/main/java/seedu/address/ui/PropertyListPanel.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+     */
+    public PropertyListPanel(ObservableList<Property> propertyList) {
+        super(FXML);
+        propertyListView.setItems(propertyList);
+        propertyListView.setCellFactory(listView -> new PropertyListViewCell());
+    }
+
+    /**
+     * Custom {@code ListCell} that displays the graphics of a {@code Person} using a {@code PersonCard}.
```
Comment (jchilling) [inline src/main/java/seedu/address/ui/PropertyListPanel.java:34]:
```suggestion
     * Custom {@code ListCell} that displays the graphics of a {@code Property} using a {@code PropertyCard}.
```

**Review by jchilling — APPROVED** [review]
Overall looks great! Just some JavaDoc changes 👍

File: src/main/java/seedu/address/model/AddressBook.java (hunk 3 above)
Comment (chngchngchng) [reply src/main/java/seedu/address/model/AddressBook.java:155]: oh yes, i was checking where the code was throwing me a NullPointerException.

**Pull request merged**